libs/agentc_cli/agentc_cli/cmds/clean.py

```
@@ -1,43 +1,66 @@
+from pydantic import ValidationError
 from typing_extensions import Literal
+from tzlocal import get_localzone
 
 logger = logging.getLogger(__name__)
 
 
-def clean_local(ctx: Context | None, type_metadata: str):
-    if type_metadata == "catalog":
-        default_folder = [DEFAULT_CATALOG_FOLDER]
-    elif type_metadata == "activity":
-        default_folder = [DEFAULT_ACTIVITY_FOLDER]
-    else:
-        default_folder = [DEFAULT_ACTIVITY_FOLDER, DEFAULT_CATALOG_FOLDER]
+def clean_local(ctx: Context | None, type_metadata: str, date: str = None):
+    clean_catalog = type_metadata == "catalog" or type_metadata == "all"
+    clean_activity = type_metadata == "activity" or type_metadata == "all"
+
+    if clean_catalog:
+        remove_directory(ctx.catalog)
+
+    if clean_activity:
+        if date is not None:
+            req_date = dateparser.parse(date)
+            local_tz = get_localzone()
+            req_date = req_date.replace(tzinfo=local_tz)
+
+            if req_date is None:
+                raise ValueError(f"Invalid date provided: {date}")
+
+            log_path = pathlib.Path(ctx.activity) / DEFAULT_LLM_ACTIVITY_NAME
+            try:
+                with log_path.open("r+") as fp:
+                    # read all lines before editing the file
+                    lines = fp.readlines()
+                    # move file pointer to the beginning of a file
+                    fp.seek(0)
+                    for line in lines:
+                        try:
+                            cur_log = Log.model_validate_json(line.strip())
+                            cur_log_timestamp = dateparser.parse(cur_log.timestamp.isoformat())
+                            if cur_log_timestamp >= req_date:
+                                fp.write(line)
+                        except ValidationError as e:
+                            logger.error(f"Invalid log entry: {e}")
+                    # truncate the file
+                    fp.truncate()
```

log files also get compressed and rotated when they get too big. you can raise a NotImplemented exception if you find that the given date exceeds the log file currently being written to, because at that point you would need to a) uncompress the previously used log file, b) scan the file to find a satisfying entry, and c) potentially recompress the file.

if you've ever had to implement a B+ tree, think back to the types of issues you ran into there :-)

---

will raise an Exception or show warning that no logs before the given date are found locally